src/libYARP_OS/src/Contact.cpp

```diff
@@ -197,6 +198,87 @@ Contact Contact::fromString(const ConstString& txt)
+}
+
+bool Contact::isValidRegisteredName(const char* name)
+{
+    const std::string port_name_s(name);
+    std::regex port_exp(
+        "(?:([a-zA-Z0-9]+(?:[_][a-zA-Z0-9]+)*):/)?"         // carrier_name:/   - optional
+        "("
+            "("                                             // /port[category]@/node~wire syntax
+                "((?:/[a-zA-Z0-9]+(?:[_:][a-zA-Z0-9]+)*)+)" // port_name        - necessary
+                "(?:"
+                    "([\\+|-][1]?)?"                        // category         - optional, "+" | "-" | "+1" | "-1"
+                    "(@((?:/[a-zA-Z0-9]+(?:[_:][a-zA-Z0-9]+)*)*))"
+                                                            // @node_name       - optional, could also be ""
+                    "(~([_a-zA-Z0-9]+))?"                   // ~wire_type       - optional
+                ")?"
+            ")"
+            "|"                                             // OR
+            "("                                             // /node[category]#/port~wire syntax
+                "((?:/[a-zA-Z0-9]+(?:[_:][a-zA-Z0-9]+)*)+)" // node_name        - necessary
+                "([\\+|-][1]?)?"                            // category         - optional, "+" | "-" | "+1" | "-1"
+                "\x23"                                      // #                - necessary
+                "((?:/[a-zA-Z0-9]+(?:[_:][a-zA-Z0-9]+)*)+)" // /port_name       - necessary
+                "(~([_a-zA-Z0-9]+))?"                       // ~wire_type       - optional
+            ")"
+            "|"                                             // OR
+            "("                                             // /node=[category]/port~wire syntax
+                "((?:/[a-zA-Z0-9]+(?:[_:][a-zA-Z0-9]+)*)+)" // /node_name       - necessary
+                "="                                         // =                - necessary
+                "([\\+|-][1]?)?"                            // category         - optional, "+" | "-" | "+1" | "-1"
+                "((?:/[a-zA-Z0-9]+(?:[_:][a-zA-Z0-9]+)*)+)" // /port_name       - necessary
+                "(~([_a-zA-Z0-9]+))?"                       // ~wire_type       - optional
+            ")"
+            "|"                                             // OR
+            "(/"                                            // "/machine:NNN/" syntax
+                "[a-zA-Z0-9]+(?:\\.[a-zA-Z0-9]+)*"          // machine name     - "machine.domain.name" or IP addr
+                ":[\\d]+/?"                                 // port number
+            ")"
+        ")");
+
+    // Matching
+    std::smatch port_match;
+    bool match = std::regex_match(port_name_s, port_match, port_exp);
+
+    if (match)
+    {
+        return true;
+    }
+    else // basic error reporting:
+    {
+        std::cout << "\"" << port_name_s << "\" does not match the correct pattern for port definition.\n";
```

Comment:
Given that this is an error, perhaps it is better to use `std::cerr` or one of the countless logging mechanism that can be used in `libYARP_OS` ?